File: ebanking/lib/models/jwt_response.dart
class JwtResponse {
  final String       token;
  final String       type;
  final int?         id;
  final String       username;
  final String?      email;
  final bool         needs2FA;
  final List<String> roles;

  JwtResponse({
    required this.token,
    required this.type,
    required this.username,
    this.id,
    this.email,
    this.needs2FA = false,
    this.roles    = const [],
  });

  /// Rôle principal (le premier dans la liste)
  String get primaryRole => roles.isNotEmpty ? roles.first : 'ROLE_CLIENT';

  bool get isAdmin  => roles.contains('ROLE_ADMIN');
  bool get isTeller => roles.contains('ROLE_TELLER');
  bool get isClient => roles.contains('ROLE_CLIENT');

  factory JwtResponse.fromJson(Map<String, dynamic> json) {
    return JwtResponse(
      token:    json['token']    as String? ?? '',
      type:     json['type']     as String? ?? 'Bearer',
      id:       json['id']       != null ? (json['id'] as num).toInt() : null,
      username: json['username'] as String? ?? '',
      email:    json['email']    as String?,
      needs2FA: json['needs2FA'] as bool?   ?? false,
      roles:    (json['roles']   as List?)?.map((e) => e.toString()).toList() ?? [],
    );
  }
}

File: src/main/java/com/bank/ebanking/dto/JwtResponse.java
package com.bank.ebanking.dto;

import java.util.List;

public class JwtResponse {

    private String       token;
    private String       type     = "Bearer";
    private Long         id;
    private String       username;
    private String       email;
    private boolean      needs2FA;
    private List<String> roles;   // ← ROLE_CLIENT / ROLE_TELLER / ROLE_ADMIN

    public JwtResponse(String token, Long id, String username,
                       String email, boolean needs2FA, List<String> roles) {
        this.token    = token;
        this.id       = id;
        this.username = username;
        this.email    = email;
        this.needs2FA = needs2FA;
        this.roles    = roles;
    }

    public String       getToken()    { return token;    }
    public String       getType()     { return type;     }
    public Long         getId()       { return id;       }
    public String       getUsername() { return username; }
    public String       getEmail()    { return email;    }
    public boolean      isNeeds2FA()  { return needs2FA; }
    public List<String> getRoles()    { return roles;    }
}